libusb/hid.c

```diff
@@ -548,11 +551,100 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
+
+static void fill_device_info_for_device(libusb_device_handle *handle, struct hid_device_info * cur_dev, struct libusb_device_descriptor *desc)
```

Comment:
general comment: can we make it in the same form as on macOS?
I.e.:
`static struct hid_device_info * fill_device_info_for_device(libusb_device_handle *handle, struct libusb_device_descriptor *desc)`
This way it would make the usage a bit simpler:
- no need to allocate `hid_device_info` at every usage (2 places);
- simpler future development: libusb backend should probably report multiple devices for multiple usages as it is done for other backends;

Reply:
I believe this way it will be possible to make the lazy initialisation with simple implementation

Reply:
Good point. I have made those changes.
Lazy initialisation is still tricky, as I don't see where `interface_num` and `bConfigurationValue` are being stored, which are both needed to reform the path. It is possible they are there and I havent spotted them, or maybe they can be retrieved from the open libusb handle?